scripts/sfpshow

```diff
@@ -0,0 +1,225 @@
+        out_put = out_put + sfp_info_output
+
+        if dump_dom:
+            sfp_type = sfp_info_dict['type']
+            dom_info_dict = state_db.get_all(state_db.STATE_DB, 'TRANSCEIVER_DOM_SENSOR|{}'.format(interface_name))
+            dom_output = self.convert_dom_to_output_string(sfp_type, dom_info_dict)
+            out_put = out_put + dom_output
+
+        return out_put
+
+    def display_eeprom(self, interfacename, dump_dom):
+        out_put = ''
+
+        if interfacename is not None:
+            presence = self.sdb.exists(self.sdb.STATE_DB, 'TRANSCEIVER_INFO|{}'.format(interfacename))
+            if presence:
+                out_put = self.convert_interface_sfp_info_to_cli_output_string(self.sdb, interfacename, dump_dom)
+            else:
+               out_put = out_put + interfacename + ': ' + 'SFP EEPROM Not detected' + '\n' 
+        else:
+            port_table_keys = self.adb.keys(self.adb.APPL_DB, "PORT_TABLE:*")
+            sorted_table_keys = natsorted(port_table_keys)
+            for i in sorted_table_keys:
+                interface = re.split(':', i, maxsplit=1)[-1].strip()
+                if interface and interface.startswith('Ethernet'):
+                    presence = self.sdb.exists(self.sdb.STATE_DB, 'TRANSCEIVER_INFO|{}'.format(interface))
+                    if presence:
+                        out_put = out_put + self.convert_interface_sfp_info_to_cli_output_string(self.sdb, interface, dump_dom)
+                    else:
+                       out_put = out_put + interface + ': ' + 'SFP EEPROM Not detected' + '\n' 
+
+                out_put = out_put + '\n' 
+
+        print out_put
+
+    def display_presence(self, interfacename):
+        port_table = []
+        header = ['Port', 'Presence']
+
+        if interfacename is not None:
+            presence = self.sdb.exists(self.sdb.STATE_DB, 'TRANSCEIVER_INFO|{}'.format(interfacename))
+            if presence:
+                port_table.append((interfacename, 'Present'))
+            else:
+                port_table.append((interfacename, 'Not present'))
+        else:
+            port_table_keys = self.adb.keys(self.adb.APPL_DB, "PORT_TABLE:*")
+            for i in port_table_keys:
+                key = re.split(':', i, maxsplit=1)[-1].strip()
+                if key and key.startswith('Ethernet'):
+                    presence = self.sdb.exists(self.sdb.STATE_DB, 'TRANSCEIVER_INFO|{}'.format(key))
+                    if presence:
+                        port_table.append((key,'Present'))
+                    else:
+                        port_table.append((key,'Not present'))
+        
+        sorted_port_table = natsorted(port_table)
+        click.echo(tabulate(sorted_port_table, header))
+
+# This is our main entrypoint - the main 'sfpshow' command
+@click.group()
+def cli():
+    """sfpshow - Command line utility for display SFP transceivers information"""
+    pass
+
+# 'show' subgroup
+@cli.group()
+def show():
```

Comment:
Since this new utility is called "sfpshow", if its only purpose is to show (i.e., read, not write) data, we can remove this redundant "show" group, simplifying the commands to `sfpshow presence` etc.

Reply:
@jleveque changed.